tests/models/nextflow/test_nextflow_deliver.py

```diff
@@ -1,45 +1,110 @@
-from typing import Dict, Optional
+from pathlib import Path
 
 import pytest
 from pydantic.v1 import ValidationError as PydanticValidationError
 
-from cg.models.nf_analysis import NextflowDeliverables
+from cg.models.nf_analysis import FileDeliverable
 
 
-def test_instantiate_nextflow_deliverables(
-    nextflow_deliverables: Dict[str, str],
+def test_file_deliverables(deliverable_id: str, file_format: str, step: str, existing_file: Path):
+    """Tests file delivery."""
+    # GIVEN valid deliverables fields
+
+    # WHEN instantiating a deliverables object
+    file_deliverable: FileDeliverable = FileDeliverable(
+        format=file_format,
+        id=deliverable_id,
+        path=existing_file,
+        path_index=existing_file,
+        step=step,
+        tag=step,
+    )
+
+    # THEN assert that it was successfully created
+    assert isinstance(file_deliverable, FileDeliverable)
+
+
+def test_file_deliverables_missing_optional(
+    deliverable_id: str, file_format: str, step: str, existing_file: Path
 ):
-    """
-    Tests nextflow delivery object.
-    """
-    # GIVEN a sample with fastq files and strandedness
+    """Tests file delivery when an optional field is missing."""
+    # GIVEN valid deliverables fields
 
     # WHEN instantiating a deliverables object
-    nextflow_deliverables_object: NextflowDeliverables = NextflowDeliverables(
-        deliverables=nextflow_deliverables
+    file_deliverable: FileDeliverable = FileDeliverable(
```

Comment:
- We have agreed to not use type hints when calling a constructor.
```suggestion
    file_deliverable = FileDeliverable(
```
This is clear enough 👍